packages/dd-trace/src/encode/agentless-ci-visibility.js

```diff
@@ -198,7 +198,14 @@ class AgentlessCiVisibilityEncoder extends AgentEncoder {
 
   _encode (bytes, trace) {
     this._eventCount += trace.length
-    const events = trace.map(formatSpan)
+    const rawEvents = trace.map(formatSpan)
+
+    const testSessionEvents = rawEvents.filter(
+      event => event.type === 'test_session_end' || event.type === 'test_suite_end'
+    )
+
+    const isTestSessionTrace = !!testSessionEvents.length
+    const events = isTestSessionTrace ? testSessionEvents : rawEvents
```

Comment:
Can this be tested?

Reply:
indeed it can be. And good thing that you brought it up because I found a nasty bug 😅

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/2431/commits/72f64280463d34fe89af2d5e65fe109d42df69a6